pynumdiff/tests/test_optimize.py

```diff
@@ -1,157 +1,118 @@
-"""
-Unit tests for optimization module
-"""
-# pylint: skip-file
 import numpy as np
-import pytest 
+from pytest import mark, skip
 
-from pynumdiff.optimize.finite_difference import first_order
-from pynumdiff.optimize.smooth_finite_difference import mediandiff, meandiff, gaussiandiff, \
-    friedrichsdiff, butterdiff, splinediff
-from pynumdiff.optimize.total_variation_regularization import *
-from pynumdiff.optimize.linear_model import *
-from pynumdiff.optimize.kalman_smooth import constant_velocity, constant_acceleration, \
-    constant_jerk
-from pynumdiff.utils.simulate import pi_control
+from ..finite_difference import first_order as iterated_finite_difference # actually second order
```

Comment:
We now import the functions *themselves*, not identically-named shadows with different parameters. If we don't give extra search space information to the optimizer, it can fish that information out of its memory. `bounds` are enforced.